File: 백준/Silver/25206. 너의 평점은/너의 평점은.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func gToGp(grade string) float64 {
	result := 0.0
	switch grade {
	case "A+":
		result = 4.5
	case "A0":
		result = 4.0
	case "B+":
		result = 3.5
	case "B0":
		result = 3.0
	case "C+":
		result = 2.5
	case "C0":
		result = 2.0
	case "D+":
		result = 1.5
	case "D0":
		result = 1.0
	case "F":
		result = 0.0
	}
	return result
}

func main() {
	r := bufio.NewReader(os.Stdin)
	w := bufio.NewWriter(os.Stdout)

	totalGp := 0.0
	sumGpTimesG := 0.0

	for i := 0; i < 20; i++ {
		strInput, _ := r.ReadString('\n')
		gradeInfo := strings.Split(strings.TrimSpace(strInput), " ")
		gp, _ := strconv.ParseFloat(gradeInfo[1], 64)
		g := gradeInfo[2]

		if g != "P" {
			sumGpTimesG += (gp * gToGp(g))
			totalGp += gp
		}
	}
	fmt.Fprint(w, sumGpTimesG/totalGp)
	w.Flush()
}

File: 백준/Silver/25206. 너의 평점은/너의 평점은.py
total_gp = 0
sum_gp_times_g = 0

def g_to_gp(grade):
    if grade == "A+":
        gp = 4.5
    elif grade == "A0":
        gp = 4.0
    elif grade == "B+":
        gp = 3.5
    elif grade == "B0":
        gp = 3.0
    elif grade == "C+":
        gp = 2.5
    elif grade == "C0":
        gp = 2.0
    elif grade == "D+":
        gp = 1.5
    elif grade == "D0":
        gp = 1.0
    elif grade == "F":
        gp = 0.0
    return gp

for n in range(20):
    sub, gp, g = input().split()
    if g == "P":
        pass
    else:
        sum_gp_times_g+=(float(gp) * g_to_gp(g))
        total_gp+=float(gp)


print(sum_gp_times_g/total_gp)